runtime/src/runtime.lisp

```diff
@@ -60,17 +60,29 @@
 (defun random-id (len)
   (map 'string #'code-char (loop for i from 1 to len collecting (+ 97 (random 26)))))
 
-(defun run-bot (command args)
-  (sb-ext:run-program command args :wait nil :input :stream :output :stream :search t))
+(defun run-bot (command args error-stream)
+  (sb-ext:run-program command args
+                      :wait nil
+                      :input :stream
+                      :output :stream
+                      :error error-stream
+                      :search t))
 
 (defmethod initialise-bot ((bot concrete-bot) log-stream)
-  (let ((initialised-bot (make-instance 'concrete-bot
-                                        :bot-process (run-bot "bash" (list "-c" (command bot)))
-                                        :bot-id (bot-id bot)
-                                        :bot-name (bot-name bot)
-                                        :bot-definition (bot-definition bot)
-                                        :command (command bot)
-                                        :bot-restarts (+ (bot-restarts bot) 1))))
+  (let* ((bot-id (random-id 10))
+         (bot-name (-> (bot-definition bot) name))
+         (initialised-bot (make-instance 'concrete-bot
+                                         :bot-process (run-bot "bash"
+                                                               (list "-c" (command bot))
+                                                               (open (format nil "~a-~a.err" bot-name bot-id)
```

Comment:
This is a dangling file handle. I think the bot error file stream should be provided by the the game-runner. An additional stream like the log-stream (perhaps error-stream) could work.